faculty/clients/job.py

```diff
@@ -125,6 +154,93 @@ def make_job_summary(self, data):
         return JobSummary(**data)
 
 
+class InstanceSizeSchema(BaseSchema):
+    milli_cpus = fields.Integer(data_key="milliCpus", required=True)
+    memory_mb = fields.Integer(data_key="memoryMb", required=True)
+
+    @post_load
+    def make_instance_size(self, data):
+        return InstanceSize(**data)
+
+
+class JobParameterSchema(BaseSchema):
+    name = fields.String(required=True)
+    type = EnumField(
+        ParameterType, by_value=True, required=True
+    )
+    default = fields.String(required=True)
+    required = fields.Boolean(required=True)
+
+    @post_load
+    def make_job_parameter(self, data):
+        return JobParameter(**data)
+
+
+class JobCommandSchema(BaseSchema):
+    name = fields.String(required=True)
+    parameters = fields.List(fields.Nested(JobParameterSchema), required=True)
+
+    @post_load
+    def make_job_command(self, data):
+        return JobCommand(**data)
+
+
+class JobDefinitionSchema(BaseSchema):
+    working_dir = fields.String(data_key="workingDir", required=True)
+    command = fields.Nested(JobCommandSchema, required=True)
+    image_type = EnumField(
+        ImageType, by_value=True, data_key="imageType", required=True
+    )
+    conda_environment = fields.String(
+        data_key="condaEnvironment", missing=None
+    )
+    environment_ids = fields.List(
+        fields.String(), data_key="environmentIds", required=True
+    )
+    instance_size_type = fields.String(
+        data_key="instanceSizeType", required=True
+    )
+    instance_size = fields.Nested(InstanceSizeSchema, data_key="instanceSize")
```

Comment:
Not sure I understand this comment - I've added tests to check that the instance size is set if and  only if instance_size_type is custom. Are you saying if we pass something like "m4.xlarge" as instace_size_type, deserialization will fail right now?

Reply:
Yes, I believe that is true.

The tests you've added to check these invalid cases are great - what I'm suggesting is to add an additional test for the other "happy path" where:

* `instance_size_type` is something other than custom
* `instance_size` is unset

If I'm right, this will currently fail, and we should then add `missing=None` to this line to fix.